recipes/lora_finetune_distributed.py

```diff
@@ -451,22 +468,22 @@ def save_checkpoint(
         intermediate_checkpoint = epoch + 1 < self.total_epochs
         # To prevent GPU memory from spiking during checkpoint save,
         # we consolidate the full model and optim state dicts on CPU for rank 0
-        with FSDP.state_dict_type(
-            self._model,
-            StateDictType.FULL_STATE_DICT,
-            FullStateDictConfig(offload_to_cpu=True, rank0_only=True),
-            FullOptimStateDictConfig(offload_to_cpu=True, rank0_only=True),
-        ):
-            cpu_state_dict = self._model.state_dict()
-            if intermediate_checkpoint:
-                opt_state_dict = FSDP.optim_state_dict(self._model, self._optimizer)
```

Comment:
@awgu do we have examples for FSDP2 optim static dict ?

Reply:
FSDP2 optimizer state dict will also give a sharded optimizer state dict (tensors should be `DTensor`s). If we want to save/load full optimizer state dicts, then similar conversion is required as model state dict.

Reply:
got you. will try to do similar conversion for optimizer state dict